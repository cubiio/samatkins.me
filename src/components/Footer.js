import React, { Component } from 'react'
import styled from 'styled-components'

import { NAV_ANCHOR, NAV_ANCHOR_HOVER } from '../lib/theme/colours'

class Footer extends Component {
  render () {
    return (
      <FooterWrapper>
        <FooterContainer>
          <p>Built with ♥ by Sam. View source <a href='https://github.com/cubiio/samatkins.me'>code</a>. © 2016-2017 Sam Atkins</p>
        </FooterContainer>
      </FooterWrapper>
    )
  }
}

export default Footer

/*
Styles
 */

const FooterWrapper = styled.div`
  display: flex;
  justify-content: center;
  padding: 1em;
`

const FooterContainer = styled.div`
  display: flex;
  justify-content: center;
  padding-top: 1em;

  a {
    color: ${NAV_ANCHOR};
    text-decoration: none;
  }

  a:hover {
    color: ${NAV_ANCHOR_HOVER}
  }
`
