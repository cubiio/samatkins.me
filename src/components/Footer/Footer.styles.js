import styled from 'styled-components';
import { NAV_ANCHOR, NAV_ANCHOR_HOVER } from '../../lib/theme/colours';

export const FooterWrapper = styled.div`
  display: flex;
  justify-content: center;
  padding: 1em;
  overflow: auto;

  @media (max-width: 700px) {
    text-align: center;
  }
`;

export const FooterContainer = styled.div`
  display: flex;
  justify-content: center;
  padding-top: 1em;

  a {
    color: ${NAV_ANCHOR};
    text-decoration: none;
  }

  a:hover {
    color: ${NAV_ANCHOR_HOVER};
  }
`;
