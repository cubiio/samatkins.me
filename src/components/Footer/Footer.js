import React from 'react';
import * as Styles from './Footer.styles';

const Footer = () => {
  const currentYear = new Date().getFullYear();
  return (
    <Styles.FooterContainer>
      <Styles.FooterDiv>
        <p>
          Built with
          <a
            href="https://reactjs.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            {' '}
            React{' '}
          </a>
          and
          <a
            href="https://www.gatsbyjs.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            {' '}
            Gatsby
          </a>
          , deployed with
          <a
            href="https://www.netlify.com/"
            target="_blank"
            rel="noopener noreferrer"
          >
            {' '}
            Netlify
          </a>
          . Â© 2016-{currentYear} Sam Atkins
        </p>
      </Styles.FooterDiv>
    </Styles.FooterContainer>
  );
};

export default Footer;
