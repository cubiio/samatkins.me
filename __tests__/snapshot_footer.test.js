/* global it expect */

import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../src/components/Footer';

it('renders correctly', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
