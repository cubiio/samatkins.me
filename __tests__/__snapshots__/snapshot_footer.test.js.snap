// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa kpnQuw"
>
  <div
    className="sc-bwzfXH hfUZzc"
  >
    <p>
      Built with ♥ by Sam. View site source
      <a
        href="https://github.com/cubiio/samatkins.me"
        target="_blank"
      >
         code
      </a>
      . © 2016-
      2017
       Sam Atkins
    </p>
  </div>
</div>
`;
